cli/internal/run/real_run.go

```diff
@@ -97,10 +104,54 @@ func RealRun(
 		Concurrency: rs.Opts.runOpts.Concurrency,
 	}
 
-	mu := sync.Mutex{}
+	taskCount := len(engine.TaskGraph.Vertices())
```

Comment:
I can not think of any problems. Since I have also run turbo over 100 times with logging grouped I highly doubt there would be any significant problems related to this.